### DISCUSSION TITLE
OKD 4.7 console and authentication cluster operators don't become available

### DISCUSSION THREAD

**Discussion by hkayabilisim** [okd-project/okd#761] [Q&A] [answered]
**Describe the bug**
At the end of the OKD 4.7 [platform agnostic installation](https://docs.okd.io/latest/installing/installing_platform_agnostic/installing-platform-agnostic.html), console and authentication operators never become available:

     $ oc get nodes
     NAME                        STATUS   ROLES    AGE     VERSION
     master0.domain.basedomain   Ready    master   5h3m    v1.20.0+87cc9a4-1079
     master1.domain.basedomain   Ready    master   4h46m   v1.20.0+87cc9a4-1079
     master2.domain.basedomain   Ready    master   4h21m   v1.20.0+87cc9a4-1079
     worker0.domain.basedomain   Ready    worker   66m     v1.20.0+87cc9a4-1079
     worker1.domain.basedomain   Ready    worker   53m     v1.20.0+87cc9a4-1079
 
     $ oc get co
     NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE
     authentication                                                             False       True          True       4h59m
     baremetal                                  4.7.0-0.okd-2021-07-03-190901   True        False         False      4h59m
     cloud-credential                           4.7.0-0.okd-2021-07-03-190901   True        False         False      5h9m
     cluster-autoscaler                         4.7.0-0.okd-2021-07-03-190901   True        False         False      4h58m
     config-operator                            4.7.0-0.okd-2021-07-03-190901   True        False         False      4h59m
     console                                    4.7.0-0.okd-2021-07-03-190901   False       True          True       61m
     csi-snapshot-controller                    4.7.0-0.okd-2021-07-03-190901   True        False         False      4h15m
     dns                                        4.7.0-0.okd-2021-07-03-190901   True        False         False      4h57m
     etcd                                       4.7.0-0.okd-2021-07-03-190901   True        False         False      4h20m
     image-registry                             4.7.0-0.okd-2021-07-03-190901   True        False         False      4h16m
     ingress                                    4.7.0-0.okd-2021-07-03-190901   True        False         True       61m
     insights                                   4.7.0-0.okd-2021-07-03-190901   True        False         False      4h53m
     kube-apiserver                             4.7.0-0.okd-2021-07-03-190901   True        False         False      4h15m
     kube-controller-manager                    4.7.0-0.okd-2021-07-03-190901   True        False         False      4h55m
     kube-scheduler                             4.7.0-0.okd-2021-07-03-190901   True        False         False      4h55m
     kube-storage-version-migrator              4.7.0-0.okd-2021-07-03-190901   True        False         False      60m
     machine-api                                4.7.0-0.okd-2021-07-03-190901   True        False         False      4h56m
     machine-approver                           4.7.0-0.okd-2021-07-03-190901   True        False         False      4h58m
     machine-config                             4.7.0-0.okd-2021-07-03-190901   True        False         False      4h24m
     marketplace                                4.7.0-0.okd-2021-07-03-190901   True        False         False      4h58m
     monitoring                                 4.7.0-0.okd-2021-07-03-190901   True        False         False      60m
     network                                    4.7.0-0.okd-2021-07-03-190901   True        False         False      4h59m
     node-tuning                                4.7.0-0.okd-2021-07-03-190901   True        False         False      4h57m
     openshift-apiserver                        4.7.0-0.okd-2021-07-03-190901   True        False         False      4h16m
     openshift-controller-manager               4.7.0-0.okd-2021-07-03-190901   True        False         False      4h56m
     openshift-samples                          4.7.0-0.okd-2021-07-03-190901   True        False         False      4h16m
     operator-lifecycle-manager                 4.7.0-0.okd-2021-07-03-190901   True        False         False      4h56m
     operator-lifecycle-manager-catalog         4.7.0-0.okd-2021-07-03-190901   True        False         False      4h56m
     operator-lifecycle-manager-packageserver   4.7.0-0.okd-2021-07-03-190901   True        False         False      4h15m
     service-ca                                 4.7.0-0.okd-2021-07-03-190901   True        False         False      4h59m
     storage                                    4.7.0-0.okd-2021-07-03-190901   True        False         False      4h59m

**Version**
UPI method on VMware ESXi, 6.7.0
Virtual Machine Hardware Version:  13
VM NIC Types: VMXNET 3
Load Balancer: F5 BIG-IP 14.1.4.1 with "Performance (Layer 4)" virtual servers
Client Version: 4.7.0-0.okd-2021-07-03-190901
Server Version: 4.7.0-0.okd-2021-07-03-190901
Kubernetes Version: v1.20.0-1085+01c9f3f43ffcf0-dirty
Fedora CoreOS: fedora-coreos-34.20210611.3.0-live.x86_64.iso

**How reproducible**
I repeated the installation two times (with OpenShiftSDN) and one time (with OVNKubernetes), and in all three cases, I experienced the same problem.

**Log bundle**
http://timewarping.org/must-gather.tar.gz

**Comment by vrutkovs** [comment, marked as answer]
```
 OAuthVersionRouteAvailable: HTTP request to "https://oauth-openshift.apps.scop.tr.asseco-see.local/healthz" failed: dial tcp 10.250.100.245:443: connect: connection refused
      OAuthRouteCheckEndpointAccessibleControllerAvailable: Get "https://oauth-openshift.apps.scop.tr.asseco-see.local/healthz": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Appears to be LB address: 10.250.100.245. Since its a UPI install you'd need to make sure its configured properly and has healthy targets, thus converting this to a discussion.

**Reply by hkayabilisim** [reply]
Thank you for pointing to LB (I have a F5 bigip). I checked again: apperantly I forgot to set "Auto Map" feature in the 443 ingress virtual server config (it was on for api and api-int). As soon as I made this change, authentication and console operators became online and the issue is resolved. Thank you again.

![image](https://user-images.githubusercontent.com/2515171/125323961-0f5f8200-e348-11eb-9df1-acff7de6d1b3.png)
